Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java

```diff
@@ -0,0 +1,79 @@
+package ca.mcgill.cooperator.model;
+
+import java.util.Set;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+@Entity
+public class ReportConfig {
+    @Id @GeneratedValue private int id;
+    private boolean requiresFile;
+    private int deadline;
+    private boolean isDeadlineFromStart;
+    private String type;
+
+    @OneToMany(
+            mappedBy = "reportConfig",
+            cascade = CascadeType.PERSIST,
+            orphanRemoval = true,
+            fetch = FetchType.EAGER)
+    @OnDelete(action = OnDeleteAction.CASCADE)
+    private Set<ReportSectionConfig> reportSectionConfigs;
+
+    /*--- Getters and Setters ---*/
+
+    public int getId() {
+        return this.id;
+    }
+
+    public boolean getRequiresFile() {
+        return this.requiresFile;
+    }
+
+    public void setRequiresFile(boolean requiresFile) {
+        this.requiresFile = requiresFile;
+    }
+
+    public int getDeadline() {
+        return this.deadline;
+    }
+
+    public void setDeadline(int deadline) {
+        this.deadline = deadline;
+    }
+
+    public boolean getIsDeadlineFromStart() {
+        return this.isDeadlineFromStart;
+    }
+
+    public void setIsDeadlineFromStart(boolean isDeadlineFromStart) {
+        this.isDeadlineFromStart = isDeadlineFromStart;
+    }
+
```

Comment:
so if the admin modifies the due date its not longer from the start?

Reply:
No, the fromStart refers to if the deadline is counting from the first day of the co-op (i.e. due 14 days from the first day of the co-op) or the last day (14 days before the last day). I could rename this if it's confusing

Reply:
oh no its ok i just didnt know what was going on